jsprit-core/src/main/java/com/graphhopper/jsprit/core/problem/solution/route/activity/BreakActivity.java

```diff
@@ -17,93 +17,49 @@
  */
 package com.graphhopper.jsprit.core.problem.solution.route.activity;
 
-import com.graphhopper.jsprit.core.problem.AbstractActivity;
 import com.graphhopper.jsprit.core.problem.Capacity;
 import com.graphhopper.jsprit.core.problem.Location;
 import com.graphhopper.jsprit.core.problem.job.Break;
-import com.graphhopper.jsprit.core.problem.job.Service;
 
-public class BreakActivity extends AbstractActivity implements TourActivity.JobActivity {
+public class BreakActivity extends InternalJobActivity {
 
-    public static int counter = 0;
+// REMARK - Balage1551 - What is it is used for?
+//    public static int counter = 0;
 
-    public double arrTime;
+// REMARK - Balage1551 - Never used and there is a duplicate implemented in JobActivity, so that could be used.
+//    public static BreakActivity copyOf(BreakActivity breakActivity) {
+//        return new BreakActivity(breakActivity);
+//    }
 
-    public double endTime;
-
-    private Location location;
-
-    private double duration;
-
-    /**
-     * @return the arrTime
-     */
-    public double getArrTime() {
-        return arrTime;
-    }
-
-    /**
-     * @param arrTime the arrTime to set
-     */
-    public void setArrTime(double arrTime) {
-        this.arrTime = arrTime;
-    }
-
-    /**
-     * @return the endTime
-     */
-    public double getEndTime() {
-        return endTime;
-    }
-
-    /**
-     * @param endTime the endTime to set
-     */
-    public void setEndTime(double endTime) {
-        this.endTime = endTime;
+    public static BreakActivity newInstance(Break aBreak) {
+        return new BreakActivity(aBreak);
     }
 
-    public static BreakActivity copyOf(BreakActivity breakActivity) {
-        return new BreakActivity(breakActivity);
+    protected BreakActivity(Break aBreak) {
+        super(aBreak, "Break", aBreak.getLocation(), aBreak.getServiceDuration(), Capacity.createNullCapacity(aBreak.getSize()));
```

Comment:
I'm still trying to understand your comment. 
I understand that the Break activity may not have location, but it isn't a problem, because the location can be setted on activity, and the copy constructor works with null as position well.